lib/elixir/lib/macro.ex

```diff
@@ -2636,6 +2636,27 @@ defmodule Macro do
     end
   end
 
+  defp dbg_ast_to_debuggable({:case, _meta, [expr, [do: clauses]]} = ast) do
+    clause_index_var = unique_var(:clause_index, __MODULE__)
+    result_var = unique_var(:result, __MODULE__)
+    expr_var = unique_var(:expr, __MODULE__)
+
+    quote do
+      unquote(expr_var) = unquote(expr)
+
+      {unquote(result_var), unquote(clause_index_var)} =
+        case unquote(expr_var) do
+          unquote(
+            Enum.with_index(clauses, fn {:->, meta, [left, right]}, i ->
+              {:->, meta, [left, {right, i}]}
+            end)
+          )
+        end
+
+      {:case, unquote_splicing([escape(ast), expr_var, clause_index_var, result_var])}
```

Comment:
(also, I use to have more variables in a previous iteration but it's not that bad in this version)

Reply:
Aaah yeah, I def think we should put it back. It's more code, but `unquote_splicing` is a bit cryptic here, agreed. You could also do

```elixir
unquote({:case, ...})
```

no?

Reply:
This isn't the same thing I think, since the AST for a tuple size > 3 is different from the tuple?

This error seems to confirm it:

```elixir
error: invalid quoted expression: {:case, {:{}, [], [:case, [line: 346], [{:{}, [], [:+, [line: 346], [1, 1]]}, [do: [{:{}, [], [:->, [line: 347], [[2], :two]]}, {:{}, [], [:->, [line: 348], [[{:{}, [], [:_other, [line: 348], nil]}], :math_is_broken]]}]]]]}, {:expr, [counter: -576460752303423324], Macro}, {:clause_index, [counter: -576460752303423388], Macro}, {:result, [counter: -576460752303423356], Macro}}

Please make sure your quoted expressions are made of valid AST nodes. If you would like to introduce a value into the AST, such as a four-element tuple or a map, make sure to call Macro.escape/1 before
  lib/elixir/test/elixir/macro_test.exs
```

We could also have `unquote({:case, [], [...]})` but it's the same as the splicing without the meta I suppose.

Anyway, I think it's fine with either individual unquotes or splicing, please let me know which one you prefer 🙂

Reply:
(I pushed [d9c9ff5](https://github.com/elixir-lang/elixir/pull/12472/commits/d9c9ff5f2f1199ca7aced0915dcad40c8eb47869) which I think looks nicer)